File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {

    let temp = 0
    let left = 0, right = height.length-1

    while(left < right){

        let curMin = Math.min(height[left], height[right])
        temp = Math.max(temp, curMin * (right-left))
        
        if(height[left] >= height[right]){
            right--
        }else{
            left++
        }
        
    }

    return temp
    
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        
        int left = 0, right = height.length-1;
        int maxArea = 0;

        while(left < right){


            int curLeft = height[left], curRight = height[right];
            int curMin = Math.min(curLeft, curRight);

            maxArea = Math.max(maxArea, curMin * (right - left));

            if(curLeft >= curRight){
                right--;
            }else{
                left++;
            }

        }

        return maxArea;

    }
}